hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemMkDir.java

```diff
@@ -185,6 +186,8 @@ public void testVerifyGetBlobProperty() throws Exception {
 
   @Test
   public void testGetPathPropertyCalledOnMkdir() throws Exception {
+    Assume.assumeTrue(getFileSystem().getAbfsStore().getPrefixMode() == PrefixMode.BLOB);
+    Assume.assumeFalse(OperativeEndpoint.isMkdirEnabledOnDFS(getConfiguration()));
```

Comment:
Lets not go with assuming for a given configuration. test should honor all configs knobs.
What can change is: Mockito.verify can have different assertion on the basis of the configs. This will help in future when some more code is added and it breaks one of the config knob flow. What you say?

Reply:
I am thinking of adding different test methods for all the possible configs and whatever changes in assertion have to happen according to them. Let me know what you think.